spoon.rb

```diff
@@ -0,0 +1,57 @@
+STDOUT.sync = true # DO NOT REMOVE
+# Solution for https://www.codingame.com/ide/puzzle/there-is-no-spoon-episode-1
+class Spoon
+  attr_accessor :width, :height, :nodes
+
+  def initialize
+    @width = gets.to_i # the number of cells on the X axis
+    @height = gets.to_i # the number of cells on the Y axis
+    @nodes = []
+    init_nodes
+  end
+
+  def init_nodes
+    n = 0
+    height.times do
+      line = gets.chomp # width characters, each either 0 or .
+      find_node(line, n)
+      n += 1
+    end
+  end
+
+  def find_node(chars, n)
+    chars.each_char.with_index do |char, i|
+      next if char == '.'
+
+      nodes.push([i, n])
+    end
+  end
+
+  def find_right_node(node)
+    nodes.select { |n| n if n[1] == node[1] && n[0] > node[0] }.min
```

Comment:
The way to use `select` is a little different - it expects a true/false statement. Meaning that you don't have to write `if` inside it. 
So instead of 
```
array.select { |n| n if some_condition(n) }
```
you should just write
```
array.select { |n| some_condition(n) }
```